commands/trends_command.py

```diff
@@ -0,0 +1,219 @@
+import matplotlib
+import requests
+import pandas as pd
+from pytrends.request import TrendReq
+from datetime import date, timedelta
+
+from .botcommand import BotCommand
+from boterror import BotError
+from botquery import BotQuery
+from config import BASE_URL
+from constants import SEND_PHOTO_ENDPOINT, PHOTO_KEY, CHAT_ID_KEY, NEW_LINE_CHAR, SPACE, UNDERSCORE, PARSE_MODE_KEY, UNDERSCORE, SPACE
+
+"""
+returns the associated response dictionary key w.r.t to a country's name
+eg. United States -> united_states
+    canada -> Canada
+"""
+def get_country_key(name: str):
+    name_formatted = name.strip().replace(SPACE, UNDERSCORE).lower()
+    if name_formatted == 'new_zealand':
+        return 'new zealand' # For some reason, New Zealand is the only country without an underscore in the response
+    if name_formatted in { 'us', 'usa', 'states', 'america' }: # For user convenience
+        return 'united_states'
+    return name_formatted
+
+"""
+Calculates the day time_delta units away from the given date.
+time_delta_str should be in the format '<number><unit>' where number is a positive integer and unit is one of d, w, m or y
+"""
+def get_initial_date(current_date: date, time_delta_str: str):
+    allowed_units_map = {
+        'd': 1,
+        'w': 7,
+        'm': 30,
+        'y': 365
+    }
+
+    try:
+        unit = time_delta_str[-1]
+        time = int(time_delta_str[:-1])
+        
+        return current_date - timedelta(days=time * allowed_units_map[unit])
+    except:
+        raise BotError(f"Invalid argument '{time_delta_str}' for option '{TrendsCommand.TIME_FLAG}'")
+
+"""
+Validates the string 'year' is a valid year.
+"""
+def validate_year_str(year_str: str):
+    try:
+        year = int(year_str)
+        if year <= 0:
+            raise
+    except:
+        raise BotError(f"Invalid year '{year_str}'")
+
+"""
+Formats a list of items into a newline-separated string with their rankings.
+eg. ["Apple", "Banana", "Cherry"] -> 
+    1. Apple
+    2. Banana
+    3. Cherry
+"""
+def format_list_items(data, key):
+    return NEW_LINE_CHAR.join([ f"{i}. {item}" for i, item in enumerate(data[key], start=1) ])
+
+"""
+Gets the TrendsReq object from the pytrends library to start fetching trends data
+"""
+def get_trends_req():
+    return TrendReq(hl="en-US", tz=360)
```

Comment:
Can we have a constant for `en-US` and also have one for default time zone? Did you ever figure out what `360` was referring to? I'm assuming it's +6 timezone since the original code you took this from has Taiwan time zone.

Reply:
The 360 refers to the timezone offset in minutes from UTC. So this represents the UTC -6 timezone (US CST). I will add constants and comments to make this more clear